File: update_urls.py
import os
import re

for root, dirs, files in os.walk('./src'):
    for file in files:
        if file.endswith('.tsx'):
            filepath = os.path.join(root, file)
            with open(filepath, 'r', encoding='utf-8') as f:
                content = f.read()
            
            # fetch('http://127.0.0.1:8000/api/leaves/') -> fetch(`${process.env.NEXT_PUBLIC_API_URL || "http://127.0.0.1:8000"}/api/leaves/`)
            content = re.sub(
                r'fetch\(\s*[\'"]http://127\.0\.0\.1:8000([^\'"]+)[\'"]\s*(.*?)\)',
                r'fetch(`${process.env.NEXT_PUBLIC_API_URL || "http://127.0.0.1:8000"}\1`\2)',
                content
            )
            
            # : 'http://127.0.0.1:8000/api/schedules/' -> : `${process.env.NEXT_PUBLIC_API_URL || "http://127.0.0.1:8000"}/api/schedules/`
            content = re.sub(
                r':\s*[\'"]http://127\.0\.0\.1:8000([^\'"]+)[\'"]',
                r': `${process.env.NEXT_PUBLIC_API_URL || "http://127.0.0.1:8000"}\1`',
                content
            )
            
            with open(filepath, 'w', encoding='utf-8') as f:
                f.write(content)
print("Done")

File: update_urls.js
const fs = require('fs');
const path = require('path');

function walk(dir) {
    let results = [];
    const list = fs.readdirSync(dir);
    list.forEach(function(file) {
        file = dir + '/' + file;
        const stat = fs.statSync(file);
        if (stat && stat.isDirectory()) { 
            results = results.concat(walk(file));
        } else { 
            if(file.endsWith('.tsx')) results.push(file);
        }
    });
    return results;
}

const files = walk('./src');
files.forEach(file => {
    let content = fs.readFileSync(file, 'utf8');
    // Replace "http://127.0.0.1:8000/api..." with \\/api...\
    // We only replace fetch calls.
    content = content.replace(/fetch\(\s*['"]http:\/\/127\.0\.0\.1:8000([^'"]+)['"]\s*([,)])/g, 
        'fetch(\\\$2');
    
    // Also replace ternary options like : "http://127.0.0.1:8000..."
    content = content.replace(/:\s*['"]http:\/\/127\.0\.0\.1:8000([^'"]+)['"]/g, 
        ': \\\');
        
    fs.writeFileSync(file, content, 'utf8');
});
console.log('Done');
